### PR TITLE
Updated String Answerformats and Added Logic Library

### PR DESCRIPTION
**Pull request by rubberdrop**
- Added xor functionality to MathQuill for String/logic type problems.
- Included additional auto-complete words to MathQuill (such as 'xor')
- Added new answerformat type to String problems: 'sexp' (set expressions): this allows for expressions such as 'A \cap B' to be entered
- Updated MathQuill and answer checking (i.e. the comparelogic function) to accommodate String/sexp type.
- Added a new library (logic.php) to improve functionality of logic problems (building truth tables, generating random logical expressions, and more...see .html documentation file)

### CODE REVIEW

File: minimize.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+uglifyjs ../../../../javascript/drawing.js -o ../../../../javascript/drawing_min.js -m -c hoist_vars=true & uglifyjs ../../../../javascript/AMhelpers2.js -o ../../../../javascript/AMhelpers2_min.js -m -c hoist_vars=true &
```
Comment (drlippman) [inline minimize.sh:1]: There already was a buildmin.sh script for this purpose.  In particular, this script references your specific directories, so isn't generally useful.

File: tinymce4/plugins/asciimath/js/ASCIIMathMLwFallback.js [hunk 2]
```diff
@@ -172,7 +172,7 @@ var AMsymbols = [
 {input:"zeta",   tag:"mi", output:"\u03B6", tex:null, ttype:CONST},
 
 //binary operation symbols
-{input:"*",  tag:"mo", output:"\u22C5", tex:"cdot", ttype:CONST},
+{input:"*",  tag:"mo", output:"\u2295", tex:"oplus", ttype:CONST},
```
Comment (drlippman) [inline tinymce4/plugins/asciimath/js/ASCIIMathMLwFallback.js:175]: ditto on earlier

**Review by drlippman — COMMENTED** [review]
Nice work in general, but several issues here.  If you want to try reworking and resubmitting, you're welcome to.  Or if I not, I can try to manually merge and fix the issues.

**Comment by drlippman** [comment]
@rubberdrop Let me know if you are planning on submitting a revised PR, or whether you'd prefer for me to try to manually merge and make fixes.

**Comment by rubberdrop** [comment]
I can make changes in about two weeks time. Sorry, finals period got in the
way.

**Comment by drlippman** [comment]
This is being closed in favor of #340

**Pull request closed without merging**